### ISSUE TITLE
Does not work with CPhpAuthManager

### ISSUE THREAD

**Issue by circulon** [cniska/yii-auth#90]
Although the docs say the DB usage is optional it is not.

example 
do not set the class in the authmanager component of the config
then try to add a Role.
The file gets written to application.data.auth.php as it should 
but then there is an exception in the AuthBehaviour like

CException
Property "CPhpAuthManager.db" is not defined. 

It seems that several of the methods are hardcoded to ONLY use the db connection
ie getAncestor, getDescendant

This effectively means that for small apps or development purposes where you may not necessarily want to add the tables to the db it is just not possible.

**Comment by cniska** [comment]
This is true, but feel free to make a PR if you have an idea on how we could also support the php-based auth manager.
